### PR TITLE
fix: Fix order of covariance transport and material preUpdate in (C)KF

### PR DESCRIPTION
**Pull request by FabianKlimpel**
The current implementation performs material interactions before the covariance matrix is transported to the surface where the interaction occurs. This leads to an overestimation of the material effect. This PR fixes the order of covariance transport and material interaction.

### CODE REVIEW

File: Core/include/Acts/Fitter/KalmanFitter.hpp [hunk 1]
```diff
@@ -637,13 +637,16 @@ class KalmanFitter {
...
-        materialInteractor(surface, state, stepper, preUpdate);
-
         // Transport & bind the state to the current surface
         auto [boundParams, jacobian, pathLength] =
             stepper.boundState(state.stepping, *surface);
 
+        // Update state and stepper with pre material effects
+        materialInteractor(surface, state, stepper, preUpdate);
+
+        // Update state and stepper with pre material effects
```
Comment (XiaocongAi) [inline Core/include/Acts/Fitter/KalmanFitter.hpp:647]: I guess the duplication of `materialInteractor` is some typo?

Comment (FabianKlimpel) [reply Core/include/Acts/Fitter/KalmanFitter.hpp]: Right, copy&paste mistake

**Review by XiaocongAi — APPROVED** [review]
Thank you  for fixing this! Should get it in soon.

**Review by XiaocongAi — APPROVED** [review]

**Comment by paulgessinger** [comment]
@FabianKlimpel can you add a small description of what this fixes? Why is this order wrong and the other order correct? I'd like to put this in the merge commit message.

**Comment by FabianKlimpel** [comment]
> @FabianKlimpel can you add a small description of what this fixes? Why is this order wrong and the other order correct? I'd like to put this in the merge commit message.

Done

**Pull request merged**